scripts/sct_run_batch.py

```diff
@@ -0,0 +1,166 @@
+                    'Either an integer greater than or equal to one '
+                    'specifying the number of cores, 0 or a negative integer '
+                    'specifying number of cores minus that number. For example '
+                    '--jobs -1 indicates run ncores - 1 jobs in parallel.',
+                    metavar=Metavar.int)
+parser.add_argument("--path-data", help='R|Setting for environment variable: PATH_DATA\n'
+                                        'Path containing subject directories in a consistent format')
+parser.add_argument('--subject-prefix', default="sub-",
+                    help='Subject prefix, defaults to "sub-" which is the prefix used for BIDS directories. '
+                    'If the subject directories do not share a common prefix, and empty string can be '
+                    'passed here.')
+parser.add_argument('--path-output', default="./",
+                    help='R|Base directory for environment variables:\n'
+                    'PATH_RESULTS=' + os.path.join('<path-output>', 'results') + '\n'
+                    'PATH_QC=' + os.path.join('<path-output>', 'QC') + '\n'
+                    'PATH_LOG=' + os.path.join('<path-output>', 'log') + '\n'
+                    'Which are output paths for results, logs, and QC')
+parser.add_argument('--include',
+                    help='Optional regex used to filter the list of subject directories. Only process '
+                    'a subject if they match the regex. Inclusions are processed before exclusions.')
+parser.add_argument('--exclude',
+                    help='Optional regex used to filter the list of subject directories. Only process '
+                    'a subject if they do not match the regex. Exclusions are processed '
+                    'after inclusions.')
+parser.add_argument('--path-segmanual', default='.',
+                    help='R|Setting for environment variable: PATH_SEGMANUAL\n'
+                    'A path containing manual segmentations to be used by the task program.')
+parser.add_argument('--itk-threads', type=int, default=1,
+                    help='R|Setting for environment variable: ITK_GLOBAL_DEFAULT_NUMBER_OF_THREADS\n'
+                    'Number of threads to use for ITK based programs including ANTs. Set to a low '
+                    'number to avoid a large increase in memory. Defaults to 1',
+                    metavar=Metavar.int)
+parser.add_argument('--task-args', default='',
+                    help='A quoted string with extra flags and arguments to pass to the task script. '
+                    'The constructed call will look like <task> <subject> <task-args>. Where <subject> is the '
+                    'base name of the subject directory. For example, if the subject path is '
+                    '/home/path-data/sub-01 then <subject> will be sub-01. `--task-args` can include any number '
+                    'of additional arguments and flags')
```

Comment:
I find it confusing to have the description of constructed call written " <task> <subject> <task-args>" because from a UX, people might think this is how they should call sct_run_batch (whereas you are describing how arguments will be passed on to the <task> script). So i suggest this modification:
```suggestion
                    help='A quoted string with extra environment variables to pass to the <task> script. '
                    'For example: sct_run_batch process_data.sh --task-args "VAR1=23 VAR2=/usr/bob"')
```

Reply:
I like the change, although `--task-args` isn't compatible with environment variable, so I'll use something like `--task-args "-foo bar -baz /qux"`. It is possible to use environment variables, so if you want we could do 

```bash
VAR=23 VAR2=/usr/bob sct_run_batch process_data.sh --task-args "-foo bar -baz /qux"
```

but I think that overcomplicates it a bit

Reply:
Followed you suggestion up to the suggestion above without the environment variables

Reply:
ahhh! i still did not understand the purpose of "--task-args", now with your example implemented in 5aff84ba1ad229960e01b8ec66e9f4b38d80aa59 i finally get it :-)